backend/models/sequenceActivities.js

```diff
@@ -0,0 +1,298 @@
+	return activities.filter(activity => {
+		const parent = activity.parent;
+
+		if (parentId) {
+			return parent && utils.uuidToString(parent) === parentId;
+		} else {
+			return !parent;
+		}
+	}).map(activity => {
+		const subActivities = getSubActivities(utils.uuidToString(activity._id), activities);
+		return  simplifyActivity(activity, subActivities);
+	});
+};
+
+const createActivitiesTree = async(account, model, sequenceId) => {
+	let activities = {};
+	const foundActivities = await db.find(account, model + ".activities",{sequenceId: utils.stringToUUID(sequenceId)}); // filter by sequenceId
+	activities = getSubActivities(null, foundActivities);
+
+	return { activities };
+};
+
+const SequenceActivities = {};
+
+SequenceActivities.getSequenceActivityDetail = async (account, model, sequenceId, activityId) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+
+	const activity = await db.findOne(account, activityCol(model), {"_id": utils.stringToUUID(activityId)});
+
+	if (!activity) {
+		throw responseCodes.ACTIVITY_NOT_FOUND;
+	}
+
+	return cleanActivityDetail(activity);
+};
+
+SequenceActivities.create = async (account, model, sequenceId, activity) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+
+	if (!activitySchema.isValidSync(activity, { strict: true })) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	if (activity.parent && !await db.findOne(account,  activityCol(model), { _id: utils.stringToUUID(activity.parent)})) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	const _id = nodeuuid();
+	activity = {...activity, sequenceId:  utils.stringToUUID(sequenceId), _id: utils.stringToUUID(_id)};
+
+	await Promise.all([
+		db.insert(account,  activityCol(model), activity),
+		FileRef.removeFile(account, model, "activities", sequenceId)
+	]);
+
+	return {...activity, _id, sequenceId: sequenceId};
+};
+
+SequenceActivities.edit = async (account, model, sequenceId, activityId, activity) => {
+	const sequence = await db.findOne(account, model + ".sequences", { _id: utils.stringToUUID(sequenceId)});
+
+	if (!sequence) {
+		throw responseCodes.SEQUENCE_NOT_FOUND;
+	}
+
+	if (!activityEditSchema.isValidSync(activity, { strict: true })) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	if (activity.parent && !await db.findOne(account, activityCol(model), { _id: utils.stringToUUID(activity.parent)})) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	const query =  {_id: utils.stringToUUID(activityId), sequenceId: utils.stringToUUID(sequenceId)};
+	const {result} = await db.update(account, activityCol(model), query, {$set: activity});
+
+	if (!result.n) {
+		throw responseCodes.ACTIVITY_NOT_FOUND;
+	}
+
+	await FileRef.removeFile(account, model, "activities", sequenceId);
```

Comment:
might be good to make this a separate function so it's clearer, like `clearActivityListCache`  or something